src/main/java/com/vinylteam/vinyl/web/servlets/ConfirmationServlet.java

```diff
@@ -0,0 +1,111 @@
+            PageGenerator.getInstance().process("confirmation-signin", attributes, response.getWriter());
+        }
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String tokenAsString = request.getParameter("token");
+        UUID token = null;
+        try {
+            token = UUID.fromString(tokenAsString);
+        } catch (IllegalArgumentException e) {
+            log.warn("Invalid email confirmation link token parameter {'token':{}}", tokenAsString, e);
+        }
+        Optional<ConfirmationToken> optionalConfirmationTokenByLinkToken = confirmationService.findByToken(token);
+        response.setContentType("text/html;charset=utf-8");
+        Map<String, String> attributes = new HashMap<>();
+        if (optionalConfirmationTokenByLinkToken.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            log.debug("Set response status to {'status':{}}", HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            attributes.put("message", "Sorry, something went wrong on out side, we're looking into it. Please, try to login again, or contact us.");
+            PageGenerator.getInstance().process("confirmation-directions", attributes, response.getWriter());
+        } else {
+            ConfirmationToken confirmationTokenByLinkToken = optionalConfirmationTokenByLinkToken.get();
+            User userByLinkToken = userService.findById(confirmationTokenByLinkToken.getUserId()).get();
+            String email = request.getParameter("email");
+            if (userByLinkToken.getEmail().equalsIgnoreCase(email)) {
+                String password = request.getParameter("password");
+                Optional<User> optionalUser = userService.signInCheck(email, password);
+                log.debug("Received a optional with User with password verification by the passed " +
+                        "email address and password {'email':{}, 'optionalUser':{}}", email, optionalUser);
+                if (optionalUser.isEmpty()) {
+                    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+                    log.debug("Set response status to {'status':{}}", HttpServletResponse.SC_BAD_REQUEST);
+                    attributes.put("token", tokenAsString);
+                    attributes.put("message", "Sorry, login or password is not correct, please check yours credentials and try again.");
+                    PageGenerator.getInstance().process("confirmation-signin", attributes, response.getWriter());
+                } else {
+                    User user = optionalUser.get();
+                    userService.update(user.getEmail(), user.getEmail(), password, user.getDiscogsUserName());
+                    confirmationService.deleteByUserId(user.getId());
+                    HttpSession session = request.getSession(true);
+                    session.setMaxInactiveInterval(60 * 60 * 5);
```

Comment:
It is looks like magic number. Please add it to application.properties and to this class fields.

Reply:
*it looks.
Okay!